src/main/java/org/folio/consortia/service/impl/SharingInstanceServiceImpl.java

```diff
@@ -118,6 +127,60 @@ public SharingInstanceCollection getSharingInstances(UUID consortiumId, UUID ins
     return result;
   }
 
+  @Override
+  @Transactional
+  public void completePromotingLocalInstance(String eventPayload) {
+    log.debug("completePromotingLocalInstance:: parameters eventPayload: {}", eventPayload);
+    try {
+      var promotingEvent = objectMapper.readValue(eventPayload, SharingInstance.class);
+
+      String centralTenantId = tenantService.getCentralTenantId();
+      String sourceTenantId = promotingEvent.getSourceTenantId();
+      String targetTenantId = promotingEvent.getTargetTenantId();
+      checkTenantsExistAndContainCentralTenantOrThrow(sourceTenantId, targetTenantId);
+
+      if (ObjectUtils.notEqual(centralTenantId, targetTenantId)) {
+        String massage = String.format("promotion failed as targetTenantId: %s does not equal to centralTenantId: %s", targetTenantId, centralTenantId);
+        log.warn("completePromotingLocalInstance:: " + massage);
+        saveInstanceWithErrorMsg(promotingEvent, massage);
+        return;
+      }
+
+      var specification = constructSpecification(promotingEvent.getInstanceIdentifier(), sourceTenantId, targetTenantId, null);
+      var optionalSharingInstance = sharingInstanceRepository.findOne(specification);
+
+      if (optionalSharingInstance.isEmpty()) {
+        String massage = String.format("sharingInstance with instanceIdentifier: %s, sourceTenantId: %s, targetTenantId: %s does not exist",
+          promotingEvent.getInstanceIdentifier(), sourceTenantId, targetTenantId);
+        log.warn("completePromotingLocalInstance:: " + massage);
+        saveInstanceWithErrorMsg(promotingEvent, massage);
+        return;
+      }
+
+      var promotedSharingInstance = optionalSharingInstance.get();
+      if (ObjectUtils.isNotEmpty(promotingEvent.getError())) {
+        promotedSharingInstance.setStatus(Status.ERROR);
+        promotedSharingInstance.setError(promotingEvent.getError());
+      } else {
+        promotedSharingInstance.setStatus(Status.COMPLETE);
+      }
+
+      sharingInstanceRepository.save(promotedSharingInstance);
+      log.info("completePromotingLocalInstance:: status of sharingInstance with instanceIdentifier: {}, sourceTenantId: {}, targetTenantId: {} " +
+        "has been updated to: {}", promotedSharingInstance.getInstanceId(), sourceTenantId, targetTenantId, promotedSharingInstance.getStatus());
+    } catch (Exception e) {
+      log.error("completePromotingLocalInstance:: exception occurred while promoting local sharing instance", e);
+      saveInstanceWithErrorMsg(new SharingInstance(), e.getMessage());
+    }
+  }
+
+  private void saveInstanceWithErrorMsg(SharingInstance sharingInstance, String errorMsg) {
+    SharingInstanceEntity sharingInstanceEntity = toEntity(sharingInstance);
```

Comment:
We should update existing sharing_instance record with error, we should not create new one

Reply:
Fixed